### ISSUE TITLE
Use of reinterpret_cast within the library

### ISSUE THREAD

**Issue by zhu3** [p12tic/libsimdpp#91]
Grepping though the library I see lots of dubious uses of reinterpret_cast and void*. Some of them are valid, but the rest of them will potentially break according to strict aliasing rules.

In some cases this leaks into public API as well:

```
template<unsigned N, class V> SIMDPP_INL
void store(void* p, const any_vec<N,V>& a)
```

Why not do something like this:

```
template<unsigned N, class V, class T> SIMDPP_INL
void store(T* p, const any_vec<N,V>& a)
```

and then reinterpret_cast the result to char*.

**Comment by p12tic** [comment]
Thanks for the bug report. The uses of `reinterpret_cast` are being already removed from the code base already, see the dev branch. Your suggestion of templating `void*` pointers in the public API makes sense, it just needs to be investigated whether this change is fully backwards-compatible.

**Comment by p12tic** [comment]
As of 331677c0b3bf6eb4d3a3d7c6a7a768c1931c8770 the library no longer contains questionable uses of `reinterpret_cast`. Suggestion to use template arguments for load and store functions has been implemented in fd1f016cfd58729e57d83917a807858d169becf0.
